provisioner/plain-v0/README.md

```diff
@@ -1,3 +1,197 @@
+manifests in a given directory. For example, below is an example Dockerfile that builds a plain `plain+v0` bundle from a
+directory containing static manifests.
+
+```dockerfile
+FROM scratch
+LABEL core.rukpak.io.bundle.mediatype.v1=plain+v0
+LABEL core.rukpak.io.bundle.manifests.v1=/manifests
+LABEL core.rukpak.io.bundle.package.v1=my-package
+LABEL core.rukpak.io.bundle.version.v1=0.0.1
+COPY ./manifests /manifests
+```
+
+where the given `manifests` directory contains the kubernetes resources required to deploy an application, for example:
+
+```
+manifests
+├── 00_namespace.yaml
+├── 01_cluster_role.yaml
+├── 02_role.yaml
+├── 03_serviceaccount.yaml
+├── 04_cluster_role_binding.yaml
+├── 05_role_binding.yaml
+└── 06_deployment.yaml
+```
+
+The `plain-v0` provisioner is able to unpack a given `plain+v0` bundle onto a cluster and then instantiate it, making
+the content of the bundle available in the cluster. It does so by reconciling `Bundle` and `BundleInstance` types that
+have the `spec.provisionerClassName` field set to `core.rukpak.io/plain-v0`. This field must be set to the correct
+provisioner name in order for the `plain-v0` provisioner to see and interact with the bundle.
+
+Below is an example of the provisioner reconciliation flow:
+
+```mermaid
+graph TD
+    C[Provisioner]
+    C -->|Watches| D[Bundle]
+    C -->|Watches| E[BundleInstance]
+    D -->|References| F[Content]
+    E -->|Creates/Manages| F[Content]
+```
+
+## Use cases
+
+### Install and apply a specific version of a bundle
+
+The `plain-v0` provisioner can install and make available a specific `plain+v0` bundle in the cluster.
+
+Simply create a `Bundle` resource pointing to a specific version of your bundle, and a `BundleInstance` which references
+that bundle. The `plain-v0` provisioner will unpack the provided Bundle onto the cluster, and eventually make the
+content available on the cluster.
+
+```yaml
+apiVersion: core.rukpak.io/v1alpha1
+kind: Bundle
+metadata:
+  name: my-bundle
+spec:
+  image: my-bundle@sha256:xyz123
+  provisionerClassName: core.rukpak.io/plain-v0
+---
+apiVersion: core.rukpak.io/v1alpha1
+kind: BundleInstance
+metadata:
+  name: my-bundle-instance
+spec:
+  provisionerClassName: core.rukpak.io/plain-v0
+  bundleName: my-bundle
+``` 
+
+First, the Bundle will be in the Pending stage as the provisioner sees it and begins unpacking the referenced content:
+
+```
+NAME          IMAGE                                        PHASE     AGE
+my-bundle     my-bundle@sha256:xyz123                      Pending   3s
+```
+
+Then eventually, as the bundle content is unpacked onto the cluster via the defined storage mechanism, the bundle status
+will be updated to Unpacked, indicating that all its contents have been stored on-cluster.
+
+```
+NAME          IMAGE                                        PHASE      AGE
+my-bundle     my-bundle@sha256:xyz123                      Unpacked   13m
```

Comment:
Does it normally take this long to unpack?

Reply:
No. This is a good point. It might make sense to do `kubectl get bundles -w` and grab the relevant lines from that output to better reflect the time it takes.

In my tests, its usually just a few seconds (but its dependent on whether the image needs to be pulled and how big the image is)

Reply:
Yeah it doesn't, it was just sitting around on my cluster for a while 😄 good catch